packages/core/src/domain/session/storeStrategies/sessionInCookie.ts

```diff
@@ -23,11 +23,17 @@ export function initCookieStrategy(cookieOptions: CookieOptions): SessionStoreSt
     isLockEnabled: isChromium(),
     persistSession: persistSessionCookie(cookieOptions),
     retrieveSession: retrieveSessionCookie,
-    clearSession: deleteSessionCookie(cookieOptions),
+    clearSession: () => setInitialSessionCookie(cookieOptions),
   }
 
   tryOldCookiesMigration(cookieStore)
 
+  const sessionCookie = retrieveSessionCookie()
+
+  if (!isSessionInitialized(sessionCookie)) {
+    setInitialSessionCookie(cookieOptions)
+  }
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏ FMU the cookieStore methods are ment to be used with the lock mechanism (aka SessionStoreOperations) to prevent cookie concurrent access issues. In here if the `setInitialSessionCookie` fails to write in the cookie, the session won't start. @bcaudan can you confirm or deny what I'm saying 🙂

You could probably  lifted up the `setInitialSessionCookie()` logic in the `sessionStore` and use the cookie `SessionStoreOperations`

Reply:
> FMU the cookieStore methods are ment to be used with the lock mechanism (aka SessionStoreOperations) to prevent cookie concurrent access issues.

+1

> You could probably lifted the setInitialSessionCookie() logic up in the sessionStore and use the cookie SessionStoreOperations

Not sure if it is exactly what you have in mind but doing that in sessionStore instead of doing it in every store strategy would be nice if possible.